judgments/tests/tests.py

```diff
@@ -4,90 +4,91 @@
+        self.assertIn("Page not found", decoded_response)
+        self.assertEqual(response.status_code, 404)
+
+
+class TestJudgmentPdfLinkText(TestCase):
+    @patch("judgments.views.detail.get_pdf_size")
+    @patch("judgments.views.detail.get_judgment_by_uri")
+    @patch.dict(environ, {"ASSETS_CDN_BASE_URL": "https://example.com"})
+    def test_pdf_link_with_size(self, mock_judgment, mock_pdf_size):
         if "ASSETS_CDN_BASE_URL" not in environ:
             raise RuntimeError("ensure ASSETS_CDN_BASE_URL is set in .env!")
-        head.return_value.headers = {}
-        head.return_value.status_code = 200
-        client.eval_xslt.return_value = "eval_xslt"
-        decoder.MultipartDecoder.from_response.return_value.parts[0].text = "part0text"
-        client.get_judgment_name.return_value = "judgment metadata"
 
-        response = self.client.get("/ewca/civ/2004/632")
+        mock_judgment.return_value = JudgmentFactory.build(is_published=True)
+        mock_pdf_size.return_value = " (1234KB)"
+
+        response = self.client.get("/test/2023/123")
         decoded_response = response.content.decode("utf-8")
-        self.assertIn(environ["ASSETS_CDN_BASE_URL"], decoded_response)
-        self.assertIn("ewca_civ_2004_632.pdf", decoded_response)
+
+        self.assertIn(
+            "https://example.com/test/2023/123/test_2023_123.pdf", decoded_response
+        )
         self.assertNotIn("data.pdf", decoded_response)
-        self.assertIn("(unknown size)", decoded_response)
-        # We don't use the Download as PDF text because there's an issue with localisated strings on CI
-        self.assertEqual(response.status_code, 200)
+        self.assertIn("(1234KB)", decoded_response)
 
-    @patch("judgments.views.detail.requests.head")
-    @patch("judgments.views.detail.decoder.MultipartDecoder")
-    @patch("judgments.views.detail.api_client")
-    def test_no_valid_pdf(self, client, decoder, head):
-        head.return_value.headers = {}
-        head.return_value.status_code = 404
-        client.eval_xslt.return_value = "eval_xslt"
-        decoder.MultipartDecoder.from_response.return_value.parts[0].text = "part0text"
-        client.get_judgment_name.return_value = "judgment metadata"
-
-        response = self.client.get("/ewca/civ/2004/632")
-        decoded_response = response.content.decode("utf-8")
-        self.assertIn("data.pdf", decoded_response)
-        self.assertNotIn("2004_632.pdf", decoded_response)
-        # We don't use the Download as PDF text because there's an issue with localisated strings on CI
-        self.assertEqual(response.status_code, 200)
+    @patch("judgments.views.detail.get_pdf_size")
+    @patch("judgments.views.detail.get_judgment_by_uri")
+    @patch.dict(environ, {"ASSETS_CDN_BASE_URL": "https://example.com"})
+    def test_pdf_link_with_no_size(self, mock_judgment, mock_pdf_size):
+        if "ASSETS_CDN_BASE_URL" not in environ:
```

Comment:
if we have patched the environ dict in the decorator, won't `"ASSETS_CDN_BASE_URL" not in environ` always be false so this error will never be raised?

Reply:
Good catch